roachtest: upgrade to/from 23.2 in mixed-version tests

renatolabs opened the pull request:
**release: support pre-releases in `PredecessorHistory`**
There was a bug computing the list of active releases within a series:
the previous logic would not take pre-releases into account, and would
just iterate over the patch releases instead. As a result, if the
latest release in a series is a pre-release, the code would
incorrectly conclude that the only existing release for that series
was the `.0` release.

In this commit, we handle the case when the latest release is a
pre-release and use that instead.

**roachtest: remove old fixture files**
Fixtures older than `v21.2` were removed. There are no tests currently
using them.

**roachtest: add v23.2 fixture files**
Generated with 23.2.0-rc.2.

**testutils: update predecessor release information**
Now that we have fixtures for 23.2, we are able to update the
predecessors file and run mixed-version tests that step through 23.2
while upgrading to master (24.1).

We also skip logictests that fail in this commit, now that the
cockroach-go config will update from 23.2 to master. These tests will
be handled separately by the owning teams.

Fixes: #115425.

Release note: None

Commit:
release: support pre-releases in `PredecessorHistory`

There was a bug computing the list of active releases within a series:
the previous logic would not take pre-releases into account, and would
just iterate over the patch releases instead. As a result, if the
latest release in a series is a pre-release, the code would
incorrectly conclude that the only existing release for that series
was the `.0` release.

In this commit, we handle the case when the latest release is a
pre-release and use that instead.

Epic: none

Release note: None

Commit:
roachtest: remove old fixture files

Fixtures older than `v21.2` were removed. There are no tests currently
using them.

Epic: none

Release note: None

Commit:
roachtest: add v23.2 fixture files

Generated with 23.2.0-rc.2.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/117581)
<!-- Reviewable:end -->

Commit:
testutils: update predecessor release information

Now that we have fixtures for 23.2, we are able to update the
predecessors file and run mixed-version tests that step through 23.2
while upgrading to master (24.1).

We also skip logictests that fail in this commit, now that the
cockroach-go config will update from 23.2 to master. These tests will
be handled separately by the owning teams.

Epic: none

Release note: None

renatolabs commented:
@cockroachdb/test-eng this is now ready for review pls 🙏

srosenberg commented on pkg/testutils/release/releases_test.go:
```diff
@@ -61,8 +68,8 @@ func TestLatestAndRandomPredecessor(t *testing.T) {
 			name:        "non-existent release",
-			v:           "v23.2.3",
-			expectedErr: `no release information for "v23.2.3" ("23.2" series)`,
+			v:           "v50.8.3",
```
Nice, we'll need to update this in 2050!

renatolabs replied:
Hehe, not really, this test is reading from `testReleaseData`, not from the real YAML file. I added a 23.2 entry in the test data to test the changes to support pre-releases, that's why I needed a new non-existent version in this test.

srosenberg reviewed (approved):
Thanks for adding the new fixtures and cleaning up the old ones!

renatolabs commented:
TFTR!

bors r=srosenberg
